### PR TITLE
Use Map in GradeList

### LINKED ISSUE #68: Change GradeList to use Map instead of List

### PR DESCRIPTION
**Pull request by juancarlovieri**
Closes #68

### CODE REVIEW

File: src/main/java/seedu/address/storage/JsonAdaptedPerson.java [hunk 1]
```diff
@@ -55,7 +57,9 @@ public JsonAdaptedPerson(@JsonProperty("name") String name, @JsonProperty("phone
             this.tags.addAll(tags);
         }
         if (grades != null) {
-            this.grades.addAll(grades);
+            this.grades = grades;
```
Comment (dominopizzaaaa) [inline src/main/java/seedu/address/storage/JsonAdaptedPerson.java:60]: does it matter it if we pass a shallow copy? is a deep copy better? (defensive programming)

Comment (juancarlovieri) [reply src/main/java/seedu/address/storage/JsonAdaptedPerson.java:60]: Lol idk man

Comment (juancarlovieri) [reply src/main/java/seedu/address/storage/JsonAdaptedPerson.java:60]: I mean we could do a new HashMap<grades>

**Review by dominopizzaaaa — APPROVED** [review]
lgtm

**Review by dominopizzaaaa — APPROVED** [review]
lgtm

**Pull request merged**